python/tests/unit/arcticdb/version_store/test_read_index.py

```diff
@@ -0,0 +1,82 @@
+"""
+Copyright 2024 Man Group Operations Limited
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt. As of the
+Change Date specified in that file, in accordance with the Business Source License, use of this software will be
+governed by the Apache License, version 2.0.
+"""
+
+import pandas as pd
+import arcticdb
+import pytest
+from functools import reduce
+
+@pytest.fixture(
+    scope="function",
+    params=(
+        "lmdb_version_store_v1",
+        "lmdb_version_store_v2",
+        "lmdb_version_store_dynamic_schema_v1",
+        "lmdb_version_store_dynamic_schema_v2",
+    ),
+)
+def lmdb_version_store_static_and_dynamic(request):
+    """
+    Designed to test all combinations between schema and encoding version for LMDB
+    """
+    yield request.getfixturevalue(request.param)
+
+@pytest.mark.skip(reason="Not implemented")
+class TestReadSingleIndex:
+    @pytest.mark.parametrize("index", [
+        pd.RangeIndex(start=0, stop=10),
+        pd.RangeIndex(start=0, stop=10, step=2),
+        pd.date_range(start="01/01/2024",end="01/10/2024"),
+        pd.MultiIndex.from_arrays(
+            [pd.date_range(start="01/01/2024", end="01/10/2024"), pd.RangeIndex(start=0, stop=10)],
+            names=["datetime", "level"]
+        )
+    ])
+    def test_rowrange(self, lmdb_version_store_static_and_dynamic, index):
+        lmdb_version_store_static_and_dynamic.write("sym", pd.DataFrame({"col": range(0, len(index))}, index=index))
+        result = lmdb_version_store_static_and_dynamic.read_index("sym")
```

Comment:
`lmdb_version_store_static_and_dynamic` is of type `NativeVersionStore`, which has a `read_index` method with different functionality.